.github/workflows/terraform_gh_runner.yml

```diff
@@ -19,40 +21,47 @@ jobs:
       - name: Set up Terraform
         uses: hashicorp/setup-terraform@v1
         with:
-          terraform_version: "1.4.2"
+          terraform_version: "1.5.7"
           cli_config_credentials_token: ${{ secrets.TF_CLOUD_TOKEN }}
 
       - name: Install dependencies
         run: |
           # Install any dependencies required by your Terraform code
 
       - name: Run Bash Script
+        env:
+          GH_TOKEN: ${{ github.token }}
         run: |
-            # GitHub repository and access token
-            repo="subspace/infra"
-            token=${{ secrets.PAT_TOKEN }}
+          # GitHub repository and access token
+          repo="subspace/infra"
+          token=${{ secrets.PAT_TOKEN }}
```

Comment:
A third token, can you explain what this third token is and why do we need it? I would have expected that workflow should be able to use `secrets.GITHUB_TOKEN`/`github.token` instead.

Reply:
PAT token for github runner registration. The `secrets.GITHUB_TOKEN` token can't be used for that purpose.

Reply:
So you're using personal token for that? Is that a recommended approach? Seems fragile and generally problematic to me.

Reply:
Yes, it's the recommended approach https://docs.github.com/en/rest/guides/getting-started-with-the-rest-api?apiVersion=2022-11-28&tool=curl#authenticating

GH_TOKEN does not have enough permissions on the repo/org level.